src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -333,6 +331,101 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 }
 
 
+/*
+ * CreateCitusLocalTablePartitionOf generates and executes the necessary commands
+ * to create a table as partition of a partitioned Citus Local Table.
+ * The conversion is done by CreateCitusLocalTable.
+ */
+void
+CreateCitusLocalTablePartitionOf(CreateStmt *createStatement, Oid relationId,
+								 Oid parentRelationId)
+{
+	if (createStatement->partspec)
+	{
+		char *parentRelationName = get_rel_name(parentRelationId);
+		char *relationName = get_rel_name(relationId);
+
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("distributing multi-level partitioned tables "
+							   "is not supported"),
+						errdetail("Relation \"%s\" is partitioned table itself "
+								  "and it is also partition of relation \"%s\".",
+								  relationName, parentRelationName)));
+	}
+
+	/*
+	 * If the parent is a citus local table, we don't need distribution column.
+	 * We can do create a Citus Local Table with current table and early return.
+	 */
+	char *attachCommand = GenerateAlterTableAttachPartitionCommand(relationId);
+	List *detachCommands = list_make3(DISABLE_DDL_PROPAGATION,
+									  GenerateDetachPartitionCommand(relationId),
+									  ENABLE_DDL_PROPAGATION);
+	ExecuteAndLogUtilityCommandList(detachCommands);
```

Comment:
Hmm, this seems to fail if there are certain foreign keys. Shouldn't we need to do something like:
```C
CascadeOperationForRelationIdList(list_make1(relationId), CASCADE_ADD_LOCAL_TABLE_TO_METADATA);
```

So that we even to attach/detach in that function?

Steps to repro the issue:
```SQL
CREATE TABLE part_table_with_very_long_name (
    dist_col integer,
    long_named_integer_col integer,
    long_named_part_col timestamp, PRIMARY KEY (long_named_part_col, dist_col)
) PARTITION BY RANGE (long_named_part_col);

CREATE TABLE part_table_with_very_long_name_p1 PARTITION OF sc1.part_table_with_very_long_name_2 FOR VALUES FROM ('2022-10-05 00:00:00') TO ('2022-10-06 00:00:00');


SELECT citus_add_local_table_to_metadata('part_table_with_very_long_name');


CREATE TABLE part_table_with_very_long_name_2 (
    dist_col integer,
    long_named_integer_col integer,
    long_named_part_col timestamp, PRIMARY KEY (long_named_part_col, dist_col)
) PARTITION BY RANGE (long_named_part_col);

CREATE TABLE part_table_with_very_long_name_2_p1 PARTITION OF part_table_with_very_long_name_2 FOR VALUES FROM ('2022-10-05 00:00:00') TO ('2022-10-06 00:00:00');


SELECT citus_add_local_table_to_metadata('part_table_with_very_long_name_2');


ALTER TABLE part_table_with_very_long_name_2 ADD CONSTRAINT fkey_cas_test_1 FOREIGN KEY (long_named_part_col, dist_col) REFERENCES part_table_with_very_long_name(long_named_part_col, dist_col);


CREATE TABLE part_table_with_very_long_name_p2 PARTITION OF part_table_with_very_long_name_2 FOR VALUES FROM ('2022-11-05 00:00:00') TO ('2022-11-06 00:00:00');
ERROR:  relation sc3.part_table_with_very_long_name_p2 is involved in a foreign key relationship with another table
```

Reply:
```sql
SELECT citus_add_node('localhost',9700,groupid=>0);
CREATE TABLE partitioned1(
    dist_col integer,
    long_named_integer_col integer,
    long_named_part_col timestamp, PRIMARY KEY (long_named_part_col, dist_col)
) PARTITION BY RANGE (long_named_part_col);

CREATE TABLE partitioned1_p1 PARTITION OF partitioned1 FOR VALUES FROM ('2022-10-05 00:00:00') TO ('2022-10-06 00:00:00');

SELECT citus_add_local_table_to_metadata('partitioned1');


CREATE TABLE partitioned2 (
    dist_col integer,
    long_named_integer_col integer,
    long_named_part_col timestamp, PRIMARY KEY (long_named_part_col, dist_col)
) PARTITION BY RANGE (long_named_part_col);

CREATE TABLE partitioned2_p1 PARTITION OF partitioned2 FOR VALUES FROM ('2022-10-05 00:00:00') TO ('2022-10-06 00:00:00');

SELECT citus_add_local_table_to_metadata('partitioned2');

ALTER TABLE partitioned2 ADD CONSTRAINT fkey_cas_test_1 FOREIGN KEY (long_named_part_col, dist_col) REFERENCES partitioned1(long_named_part_col, dist_col);

CREATE TABLE partitioned1_p2 PARTITION OF partitioned2 FOR VALUES FROM ('2022-11-05 00:00:00') TO ('2022-11-06 00:00:00');
```